src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs

```diff
@@ -649,6 +649,39 @@ internal IEnumerable<PortModel> GetGroupOutPorts(IEnumerable<NodeModel> ownerNod
                 );
         }
 
+        private double GetPortVerticalOffset(PortModel portModel)
+        {
+            // vertical offset accounts for the port margins
+            double verticalOffset = 20;
+            int index = portModel.LineIndex == -1 ? portModel.Index : portModel.LineIndex;
+
+            //If the port was not found, then it should have just been deleted. Return from function
+            if (index == -1)
+                return verticalOffset;
+
+            // calculate the vertical offset for the port index.
+            int portVerticalMidPoint = 17;
+            double portHeight = portModel.Height;
+            return verticalOffset + (index * portHeight) + portVerticalMidPoint;
```

Comment:
Will this work if there are Ports from multiple nodes on the group?

The index is relative to the owner node and not the group, right?

Reply:
Yes, you are right. Modified it to use the port index on the group.